test/unit/test_list.c

```diff
@@ -220,6 +220,53 @@ bool test_r_list_sort5(void) {
 	mu_end;
 }
 
+// To illustrate the original bug.
```

Comment:
Although we appreciate a lot unit tests, I'm not sure in this case they make much sense, as the wrong behaviour is not in the RList API, but just in the wrong implementation of the some specific sort comparators. There are already tests that check the correctness of `r_list_sort`.

Reply:
It would be good to add a comment clarifying the expected kind of comparison operator expected either next to  r_list_sort declaration or RListComparator typedef. Even though some c strandard library functions expect 3 value comparison operator, using int instead of boolean in c isn't that unusual either and 2 value comparison operator is also used by some libraries especially in C++ standard library. Even worse due to the way r_list_sort is implemented it will work correctly using 2 value comparison operator if the list has less than 43 elements, so it's easy to miss during development.

Reply:
Yes, it's exactly the 43-elements threshold why I'm adding another test. The others test on smaller lists.

I can remove the `#if FIXED || true` thing, no problem.

Adding a comment to `RListComparator` is good idea, will do.

Reply:
@ret2libc adding a unit test was helpful as an easy approach to isolate the issue, and to confirm the fix is working. It does lose that purpose when the fix is done; however, mergesorting of bigger lists isn't tested at all, so I left it there.

Reply:
Pardon; "bigger lists *of int ptrs*" that is. 52-strings list is covered in `test_r_list_sort5`.

Reply:
Ok, thanks!